src/DeepLinkModule/javasource/deeplink/utils/DeeplinkUtils.java

```diff
@@ -0,0 +1,62 @@
+package deeplink.utils;
+
+import java.util.List;
+
+import com.mendix.core.Core;
+import com.mendix.logging.ILogNode;
+import com.mendix.systemwideinterfaces.core.IContext;
+import com.mendix.systemwideinterfaces.core.IMendixObject;
+
+import deeplink.proxies.DeepLink;
+import deeplink.proxies.microflows.Microflows;
+
+public class DeeplinkUtils {
+	private static ILogNode LOG = Core.getLogger(deeplink.implementation.Commons.logNodeName);
+
+	public static void updateOrCreateDeeplink(IContext context, String deeplinkName, boolean forceLogin, String description, 
+			String microflowName, boolean keepAsHomeDeeplink,String entity, String attribute) throws Exception {
+
+		List<IMendixObject> list = Core.createXPathQuery(String.format("//%s[%s=$value]", 
+				DeepLink.getType(),
+				DeepLink.MemberNames.Name.toString()))
+				.setVariable("value",deeplinkName)
+				.execute(context);
+		
+		DeepLink deeplink = null;
+		
+		if(list.size()>=1) {
+			deeplink = DeepLink.initialize(context, list.get(0));
+			if(list.size()>1) {
+				LOG.warn(String.format("Deeplink configuration is corrupt. Multiple deep links with name '%s' found.", deeplinkName));
+			}
+		}
+		else if(list.size()==0) {
+			deeplink = new DeepLink(context);
+		}
+		
+		deeplink.proxies.Microflow microflowFromDeepLink = null;
+		
+		try {
+			microflowFromDeepLink = Microflows.dS_MicroflowByDeeplink(context, deeplink);
+			
+			deeplink.setAllowGuests(!forceLogin);
+			deeplink.setDescription(description);
+			deeplink.setMicroflow(microflowName);
+			deeplink.setName(deeplinkName);
+			deeplink.setUseAsHome(keepAsHomeDeeplink);
+			deeplink.setObjectType(entity);

```

Comment:
`entity` and `attribute` can be null. Is that ok, for `setObjectType` and `setObjectAttribute`? What if only one of them is null?
It might be good to be a bit more defensive here, and add an if-block around it, checking `entity` and `attribute` for null

Reply:
Added defensive value check.
Please check and resolve :)